### PR TITLE
mark idColumnName as deprecated

### PR DESCRIPTION
**Pull request by agnes512**
connects to https://github.com/strongloop/loopback-next/issues/4763

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-datasource-juggler) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](https://loopback.io/doc/en/contrib/style-guide-es6.html)
- [ ] Commit messages are following our [guidelines](https://loopback.io/doc/en/contrib/git-commit-messages.html)

### CODE REVIEW

File: lib/datasource.js [hunk 1]
```diff
@@ -2097,7 +2097,11 @@ DataSource.prototype.columnNames = function(modelName) {
 };
 
 /**
- * Find the ID column name
+ * @deprecated
+ * Function to get the id column name of the target model
+ *
+ * Use `SqlConnector.prototype.column(ModelDefinition.prototype.idName())` instead
```
Comment (emonddr) [inline lib/datasource.js]: Shouldn't this be:

`Connector.prototype.idMapping` ?

Comment (agnes512) [reply lib/datasource.js]: Reworded

**Comment by agnes512** [comment]
@slnode test please

**Comment by agnes512** [comment]
@slnode test please

**Comment by agnes512** [comment]
@slnode test please

**Comment by agnes512** [comment]
@slnode test please

**Pull request merged**